packages/firestore/src/core/sync_engine.ts

```diff
@@ -1043,14 +1040,15 @@ export class MultiTabSyncEngine extends SyncEngine
       // match the state on disk.
       const activeTargets = this.sharedClientState.getAllActiveQueryTargets();
       const activeQueries = await this.synchronizeQueryViewsAndRaiseSnapshots(
-        activeTargets.toArray()
+        activeTargets.toArray(),
+        /*transitionToPrimary=*/ true
       );
+      this.isPrimary = true;
+      await this.remoteStore.applyPrimaryState(true);
```

Comment:
I want to verify something. These lines of code were moved down b/c the bit should only be set if a prior IndexDB operation succeeded correct?

Reply:
Yes, exactly.